src/main/java/com/amazon/jenkins/ec2fleet/EC2RetentionStrategy.java

```diff
@@ -31,14 +33,20 @@ public class EC2RetentionStrategy extends RetentionStrategy<EC2FleetNodeComputer
     @Override
     public long check(final EC2FleetNodeComputer fc) {
         final AbstractEC2FleetCloud cloud = fc.getCloud();
+        boolean isOrphaned = false;
 
         LOGGER.fine(String.format("Checking if node '%s' is idle ", fc.getName()));
 
-        // in some multi-thread edge cases cloud could be null for some time, just be ok with that
         if (cloud == null) {
-            LOGGER.warning("Cloud is null for computer " + fc.getDisplayName()
-                    + ". This should be autofixed in a few minutes, if not please create an issue for the plugin");
-            return RE_CHECK_IN_A_MINUTE;
+            final long idleTime = System.currentTimeMillis() - fc.getIdleStartMilliseconds();
+
+            if (idleTime >= ORPHAN_TIME_LIMIT) {
+                LOGGER.log(Level.INFO, String.format("Instance %s may have been orphaned due to not having an associated cloud for too long.", fc.getDisplayName()));
+                isOrphaned = true;
+            } else {
+                LOGGER.log(Level.INFO, "Instance:" + fc.getDisplayName() + " does not currently have an associated cloud and will be marked orphaned in " + TimeUnit.MILLISECONDS.toSeconds(ORPHAN_TIME_LIMIT - idleTime) + " seconds.");
```

Comment:
It is a good idea to set a time limit on how long we allow the cloud to be null for the valid case. I remember seeing the null cloud warning once or twice before, but do you know if the `ORPHAN_TIME_LIMIT ` is enough time to allow for valid cases where cloud is not null? 
Might be a good idea to allow for at least 2 checks to happen before we consider the node orphaned?

Reply:
Yes, I think a timer for 3-5 minutes and potentially allowing users to augment the value is more reasonable.